src/test/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImplTest.java

```diff
@@ -0,0 +1,122 @@
+package com.pusher.client.channel.impl;
+
+import com.pusher.client.AuthorizationFailureException;
+import com.pusher.client.Authorizer;
+import com.pusher.client.connection.impl.InternalConnection;
+import com.pusher.client.util.Factory;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Matchers;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PrivateEncryptedChannelImplTest {
+
+    private String authorizer_valid = "{\"auth\":\"636a81ba7e7b15725c00:3ee04892514e8a669dc5d30267221f16727596688894712cad305986e6fc0f3c\",\"shared_secret\":\"iBvNoPVYwByqSfg6anjPpEQ2j051b3rt1Vmnb+z5doo=\"}";
+    private String authorizer_missingAuthKey = "{\"shared_secret\":\"iBvNoPVYwByqSfg6anjPpEQ2j051b3rt1Vmnb+z5doo=\"}";
+    private String authorizer_missingSharedSecret = "{\"auth\":\"636a81ba7e7b15725c00:3ee04892514e8a669dc5d30267221f16727596688894712cad305986e6fc0f3c\"}";
+    private String authorizer_malformedJson = "potatoes";
+
+    private InternalConnection internalConnection;
+    private Authorizer authorizer;
+    private Factory factory;
+
+    @Before
+    public void setup() {
+        internalConnection = mock(InternalConnection.class);
+        authorizer = mock(Authorizer.class);
+        factory = mock(Factory.class);
+    }
+
+    @Test
+    public void testPrepareThrowsNoExceptionIfValidAuthorizer() {
```

Comment:
```suggestion
    public void prepareSucceedsGivenValidAuthorizer() {
```
since JUnit4 there's no need to prefix tests with `test` (`@Test` annotation serves the same purpose). also there's no need to catch an exception, JUnit will automatically catch any exception and fail a test that raised an exception. `Succeeds` is simpler than `ThrowsNoException` and does not contain negation

Reply:
I kept with the test prefix because the other tests currently do, but I guess if we're leaving this in better place than when we started I can remove for this file. I also refactored the titles as we no longer have a `prepare` method but `checkAuthentication`. 

I see your point about not checking if an exception is thrown. Have removed the check for this test but kept for the others.

Reply:
Others can be also simplified using JUnit4 facility for testing expected exceptions, like[ the existing code](https://github.com/pusher/pusher-websocket-java/blob/aeb62de367a220e31dd797f37e5821ca05e278b1/src/test/java/com/pusher/client/channel/impl/ChannelImplTest.java#L55) already does.

Reply:
Hmm. I saw that, but it doesn't appear to let me assert that the error message matches what I expect. As we're using the `AuthorizationFailureException` to hold two different types of errors, I want to assert the messages match what I expect.

Reply:
I'm not sure if asserting on the message is not testing a bit two much of internal implementation details. If such bits really mattered from the perspective of the client code that should be modelled within the code design by providing a distinct exception type.

There is however a better way to do in JUnit: [the second example here](https://junit.org/junit4/javadoc/4.12/org/junit/rules/ExpectedException.html). I've checked that earlier and there seemed to be a conflict on Hamcrest dependency most likely steaming from using both Mockito and PowerMock. Maybe that could be resolved by removing PowerMock (using it is a code smell or rather working around code smells of non-testable code) or updating it to be more compatible with Mockito dependency.

Anyway for me it feels like for now it's best to don't test such details like specifics of the message. I haven't tested them in my tests.

Reply:
👍 thanks for providing more context on how to do this - I've removed testing the message strings for now. I think this would be a nice to have in the future.